src/server/pkg/obj/amazon_client.go

```diff
@@ -63,15 +74,45 @@ func (c *amazonClient) Reader(name string, offset uint64, size uint64) (io.ReadC
 		byteRange = fmt.Sprintf("bytes=%s", byteRange)
 	}
 
-	getObjectOutput, err := c.s3.GetObject(&s3.GetObjectInput{
-		Bucket: aws.String(c.bucket),
-		Key:    aws.String(name),
-		Range:  aws.String(byteRange),
-	})
-	if err != nil {
-		return nil, err
+	var reader io.ReadCloser
+	if c.usingCloudfront() {
+		var resp *http.Response
+		var connErr error
+		url := fmt.Sprintf("http://%v.cloudfront.net/%v", c.distribution, name)
+
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return nil, err
+		}
+		req.Header.Add("Range", byteRange)
+
+		backoff.RetryNotify(func() error {
+			resp, connErr = http.DefaultClient.Do(req)
+			if connErr != nil && isNetRetryable(connErr) {
+				return connErr
+			}
```

Comment:
Yup, I forgot to return `connErr` outside the retry functions. Beyond that, this version of retry doesn't use the notify function as you describe, but I've made a separate issue to standardize our usage of retry / notify: https://github.com/pachyderm/pachyderm/issues/1780